docs/benchmarks.md

```diff
@@ -1,131 +1,183 @@
+| 10MB      | 1M        | 3810                 | 33.496             | 3825.54  |
+| 50MB      | 1M        | 4370                 | 29.185             | 4426.73  |
+| 100MB     | 1M        | 3504                 | 36.421             | 3505.01  |
+| 200MB     | 1M        | 3048                 | 41.919             | 3044.43  |
+| 1GB       | 1M        | 2120                 | 60.246             | 2114.33  |
+
+## Writes
+
+### FIO spec
+
+  ```
+    [global]
+    ioengine=sync
+    direct=1
+    fadvise_hint=0
+    verify=0
+    iodepth=64
+    invalidate=1
+    time_based=0
+    file_append=0
+    # By default fio creates all files first and then starts writing to them. This option is to disable that behavior. 
+    create_on_open=1 
+    thread=1
+    openfiles=1
+    group_reporting=1
+    allrandrepeat=1
+    # Every file is written only once. Set nrfiles per thread in such a way that the test runs for 1-2 min. 
+    # This will vary based on file size. Change the value from table to get provided results.
+    nrfiles=2
+    filename_format=$jobname.$jobnum.$filenum
+    # Change this to randwrite to test random writes.
+    rw=write   
+    # Update the block size value from the table for different 
+    bs=1M
+    # Update the file size value from table(file size) for different experiments.
+    filesize=1G  
+    [experiment]
+    stonewall
+    # Change the test directory (1mb) for different experiments. The directory must exist within the mounted directory.
+    directory=gcs/1gb
+    numjobs=112
+ ```
+
+**Note:** Performance will be better if you experiment on scenarios where files
+do not exist,
+compared to scenarios where files already exist. This is because writing to an
+existing
+file requires reading the existing content first, before the new data can be
+written.
+
+### Results
+
+#### Sequential Write
+
+| File Size | BlockSize | nrfiles | Bandwidth in MiB/sec | IOPS(avg) | Avg Latency (msec) | Network Send Traffic (GiB/s) |
+|-----------|-----------|---------|----------------------|-----------|--------------------|------------------------------|
+| 256KB     | 16K       | 30      | 212                  | 14976.95  | 3.206              | 0.027                        |
+| 1MB       | 1M        | 30      | 772                  | 794.32    | 1.150              | 0.036                        |
+| 50MB      | 1M        | 20      | 3611                 | 5948.63   | 8.929              | 1.33                         |
+| 100MB     | 1M        | 10      | 3577                 | 4672.64   | 1.911              | 1.41                         |
+| 1GB       | 1M        | 2       | 1766                 | 2121.66   | 49.114             | 1.77                         |
+
+#### Random Write
+
+In case of random writes, only offset will change in calls issued by fio.
+GCSFuse behaviour will
+remain the same and there are no changes in the way gcs calls are being made.
+Hence the bandwidth will be same
```

Comment:
"In case of random writes, only offset will change in calls issued by fio. GCSfuse behavior will remain the same and there are no changes in the way gcs calls are being made.
Hence the bandwidth will be same as sequential writes"

I get this is from existing benchmark, but i have no idea what this means. Isnt this really just due to the fact we stage writes first in -temp-dir?

Reply:
Yes.. I feel so.. I checked log pattern also... pattern seems same for both random and sequential writes.

Reply:
Please change to "Random writes and sequential write performance will generally be the same, as all writes are first staged to a local temporary directory before being written to GCS on close/fsync"